webhook/handlers/patient_first_workflow.js

```diff
@@ -0,0 +1,79 @@
+const db = require('../../models')
+
+class PatientFirstController {
+
+  async bookConsulation(agent, id){
+      let fulfillmentText = "";
+      let user = await db.userWA.findOne({where: {wa_phone_number: id}});
+      let patients = await db.patientWA.findAll({where: { User_id : user.id }});
+      if(patients.length == 0){
+          fulfillmentText += ("There are no patients registered for you. Please register the patient(s) before booking the consultation.\n"
+              + "Would you like to register a patient now?\n")
+      }
+      else
+      {
+          fulfillmentText += "You have " + (patients.length).toString() + " patient" +
+          ((patients.length === 1) ? 's': '') + " registered currently:";
+          for(let i=0; i< patients.length; i++) {
+              fulfillmentText += "\nPatient " + (i+1).toString() + ": " + patients[i].first_name + " " + patients[i].last_name;
+          }
+          fulfillmentText += ("\nReply back the full name of the patient you wish to book a consult for or add a new patient.");
+      }
+      return fulfillmentText;
+  }
+
+  async patientInfo(agent, id){
+      let user = await db.userWA.findOne({where: {wa_phone_number: id}});
+      let fulfillmentText = "";
+      let paramContextName = 'patfirst_choose_patient';
```

Comment:
Probably won't need an old context for this intent. The user replies with the name they want to select right after the intent that calls `bookConsultation` function. The intent for this function just accepts the name and shows them the details of the selected patient and asks how to proceed. 
You can use: 
https://github.com/kj7kunal/E-Med/blob/develop-patient-first-workflow/webhook/handlers/user_registration_workflow.js#L222-L226

Reply:
Explicitly naming a context to extract its parameters is needed when you have some intents between the intent that accepts the required parameters and the intent that calls the function. 
For example, suppose you want to confirm the details that you have entered as a response to the `asking intent`. At that time, the `confirmation intent` finally calls the backend function. At that time, the context associated with the `asking intent` has the parameters. 
In this case, since the `asking intent` calls the backend, the current context has the parameters.